linera-base/src/crypto/secp256k1.rs

```diff
@@ -187,9 +231,13 @@ impl Secp256k1Signature {
     where
         T: BcsSignable<'de>,
     {
-        let secp = secp256k1::Secp256k1::signing_only();
-        let message = Message::from_digest(CryptoHash::new(value).as_bytes().0);
-        let signature = secp.sign_ecdsa(&message, &secret.0);
+        use k256::ecdsa::signature::hazmat::PrehashSigner;
+
+        let prehash = CryptoHash::new(value).as_bytes().0;
+        let (signature, _rid) = secret
+            .0
+            .sign_prehash(&prehash)
+            .expect("Failed to sign prehased data"); // NOTE: This is a critical error we don't control.
```

Comment:
```suggestion
            .expect("Failed to sign prehashed data"); // NOTE: This is a critical error we don't control.
```

Reply:
Done in [Apply review suggestions.](https://github.com/linera-io/linera-protocol/pull/3407/commits/8f530c26831f602c5a89abc74abc7445a02389d6).